### PR TITLE
Add dependency to docker/swarmkit

### PR DESCRIPTION
**Pull request by aluzzardi**
Add a dependency to docker/swarmkit in preparation for adding first-class Swarm-wide service management inside the Docker Engine as described in our [ROADMAP.md](https://github.com/docker/docker/blob/master/ROADMAP.md#14-cluster-capable-engine).

### CODE REVIEW

File: hack/vendor.sh [hunk 1]
```diff
@@ -59,14 +59,14 @@ clone git github.com/vdemeester/shakers 24d7f1d6a71aa5d9cbe7390e4afb66b7eef9e1b3
 clone git golang.org/x/net 2beffdc2e92c8a3027590f898fe88f69af48a3f8 https://github.com/tonistiigi/net.git
 clone git golang.org/x/sys eb2c74142fd19a79b3f237334c7384d5167b1b46 https://github.com/golang/sys.git
 clone git github.com/docker/go-units 651fc226e7441360384da338d0fd37f2440ffbe3
-clone git github.com/docker/go-connections v0.2.0
-clone git github.com/docker/engine-api 772250a752e34cacaeef7c92b8e0ddf43450b629
+clone git github.com/docker/go-connections fa2850ff103453a9ad190da0df0af134f0314b3d
+clone git github.com/docker/engine-api 6e7fae3cd9aa40e31cb5833d67a94ee8cff3054c git://github.com/vieux/engine-api.git
```
Comment (ibuildthecloud) [inline hack/vendor.sh]: I assume this won't stay as vieux's repo?

Comment (thaJeztah) [inline hack/vendor.sh]: @ibuildthecloud it should be updated after https://github.com/docker/engine-api/pull/265 is accepted and merged

Comment (thaJeztah) [inline hack/vendor.sh]: Thanks for asking though!

Comment (vdemeester) [inline hack/vendor.sh]: @ibuildthecloud definitely, there is a PR in review for that in `engine-api` :angel:

Comment (vdemeester) [inline hack/vendor.sh]: damm @thaJeztah is quick :joy_cat:

**Comment by justincormack** [comment]
This needs a rebase.

**Comment by crosbymichael** [comment]
If we get the engine-api change merged this one can be merged easily.  Its just a bunch of vendored deps

**Comment by icecrime** [comment]
In the making of this PR:
- CLI was merged into this branch from #23363 (complete history is still available on this PR)
- API and backend implementation was merged into this branch from #23362 (complete history is still available on this PR)

This branch is now updated with a squashed version of this work to avoid polluting master's history.

**Comment by icecrime** [comment]
LGTM :+1:

**Comment by crosbymichael** [comment]
LGTM

**Comment by vieux** [comment]
LGTM

**Comment by icecrime** [comment]
Ping @SvenDowideit: can you please help us with that?

```
1:04:59 ERROR: 2016/06/14 04:04:59 site.go:1211: Two or more menu items have the same name/identifier in Menu "main": "update".
21:04:59 Rename or set an unique identifier.
```

**Comment by icecrime** [comment]
Nevermind we figured it out :innocent:

**Comment by icecrime** [comment]
image

**Comment by vieux** [comment]
@mlaventure 
bumble_bee

**Comment by tonistiigi** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Comment by vdemeester** [comment]
experimal timed out after all the tests 😇

**Pull request merged**